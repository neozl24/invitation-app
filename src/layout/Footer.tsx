import heartLogo from '../assets/heart.svg'

export const Footer = () => {
  return (
    <div className='w-full h-20 sm:h-24 flex flex-col justify-center items-center border-t border-solid border-t-neutral-300'>
      <Line className='mb-1'>
        Made with
        <img src={heartLogo} alt='love icon' width='20' className='inline ml-1 mr-1' />
        in Melbourne.
      </Line>
      <Line>
        &copy;
        2016 Broccoli & Co. All rights reserved.
      </Line>
    </div>
  )
}

interface LineProps extends React.ButtonHTMLAttributes<HTMLParagraphElement> {
  children: React.ReactNode
}

const Line = ({
  className = '',
  children,
  ...props
}: LineProps) => {
  const baseClassName = 'flex items-center text-sm italic text-neutral-400'
  return (
    <p className={`${baseClassName} ${className}`} {...props}>
      {children}
    </p>
  )
}