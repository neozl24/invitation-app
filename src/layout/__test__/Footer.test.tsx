import { render } from '@testing-library/react'
import { Footer } from '../Footer'

test('renders name', () => {
  const { asFragment } = render(<Footer />)

  expect(asFragment()).toMatchSnapshot()
})
