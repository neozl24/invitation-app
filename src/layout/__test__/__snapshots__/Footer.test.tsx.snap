// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders name 1`] = `
<DocumentFragment>
  <div
    class="w-full h-16 sm:h-24 flex flex-col justify-center items-center border-t border-solid border-t-neutral-300"
  >
    <p
      class="flex items-center text-xs sm:text-sm italic text-neutral-400 mb-1"
    >
      Made with
      <img
        alt="love icon"
        class="inline ml-1 mr-1"
        src="/src/assets/heart.svg"
        width="20"
      />
      in Melbourne.
    </p>
    <p
      class="flex items-center text-xs sm:text-sm italic text-neutral-400 "
    >
      Â© 2016 Broccoli & Co. All rights reserved.
    </p>
  </div>
</DocumentFragment>
`;
